main.py

```diff
@@ -316,7 +315,7 @@ def initialize_constants(exponent, signal_length):
 # "carry_val"
 @jit
 def firstcarry(signal, power_bit_array):
-    carry = jnp.int32(0)
+    # carry = jnp.int32(0)
```

Comment:
I am assuming this is intentional, but also not sure how it can compile as it's a function without an executable body. Unless Python allows a comment, usually one must have `pass` in the function body at minimum.

Reply:
The function does have a body below this line. 🙂 Anyway, Ruff wanted to delete the variable, as it is unused, but I elected to comment it out instead, in case it is actually needed in the future...